#! /bin/sh /usr/share/dpatch/dpatch-run
## 50_glpk-sparse.dpatch by  <root@laboiss2.intra.mpipf-muenchen.mpg.de>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: New patch generated from octave2.9 2.9.4-11 diff.gz

@DPATCH@

--- octave2.9-2.9.4.orig/src/DLD-FUNCTIONS/__glpk__.cc
+++ octave2.9-2.9.4/src/DLD-FUNCTIONS/__glpk__.cc
@@ -482,7 +482,7 @@
     }
   else
     {
-      SparseMatrix A (args(1).matrix_value ()); // get the sparse matrix
+      SparseMatrix A = args(1).sparse_matrix_value (); // get the sparse matrix
 
       if (error_state)
 	{
@@ -509,7 +509,7 @@
 	    nz++;
 	    rn(nz) = A.ridx(i) + 1;
 	    cn(nz) = j + 1;
-	    a(nz) = A(i,j);
+	    a(nz) = A.data(i);
 	  }
     }
 
